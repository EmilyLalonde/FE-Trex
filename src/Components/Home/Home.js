import React, { Component } from 'react';
import { 
  StyleSheet, 
  Text, 
  View, 
  ScrollView,
  TouchableOpacity
} from 'react-native';

import WandererHeader from '../WandererHeader/WandererHeader';
import WandererFooter from '../WandererFooter/WandererFooter'

export default class Home extends Component {
  constructor(props) {
    super(props) 
      this.state = {

      }
  }

  render() {
      const {navigate} = this.props.navigation;
    return (
      <View style={styles.container}>
        <WandererHeader />
        <ScrollView>
          <Text style={styles.text}>Welcome to Ariva</Text>
            <Text style={styles.text}>I am a ...</Text>
            <TouchableOpacity>
              <Text style={styles.button}>Wanderer</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => navigate('FollowerDashboard')}>
              <Text style={styles.button}>Follower</Text>
            </TouchableOpacity>


        </ScrollView>
  
       <WandererFooter navigate={navigate} />
        
    </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000000',
  }, 
  text: {
    color: 'white',
    marginVertical: 40,
    textAlign: 'center',
    fontSize: 30,
    width: 'auto'
  }, 
  button: {
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 8,
    borderStyle: 'solid',
    width: 'auto',
    height: 60,
    margin: 20,
    fontSize: 30,
    padding: 10,
    color: 'white',
    textAlign: 'center',
    backgroundColor: '#1C4263',
    alignItems: 'stretch'
  }
});




