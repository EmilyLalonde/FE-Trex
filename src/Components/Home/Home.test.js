import React from "react";
import "react-native";
import Home from "./Home";
import { shallow } from "enzyme";

describe("Home", () => {
  it("should return true", () => {
    expect(true).toEqual(true);
  });
  it("should match the snapshot when loading", () => {
    let wrapper = shallow(<Home navigation={{ navigate: jest.fn() }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
