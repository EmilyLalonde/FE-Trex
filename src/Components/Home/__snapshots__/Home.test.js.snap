// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot when loading 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <WandererHeader />
  <ScrollViewMock>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "marginVertical": 40,
          "textAlign": "center",
        }
      }
    >
      Welcome to Trex
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 30,
          "marginVertical": 40,
          "textAlign": "center",
        }
      }
    >
      I am a ...
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "#1C4263",
            "borderColor": "white",
            "borderRadius": 8,
            "borderWidth": 1,
            "color": "white",
            "fontSize": 30,
            "height": 60,
            "margin": 20,
            "padding": 10,
            "textAlign": "center",
            "width": "auto",
          }
        }
      >
        Wanderer
      </Text>
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "#84183B",
            "borderColor": "white",
            "borderRadius": 8,
            "borderStyle": "solid",
            "borderWidth": 1,
            "color": "white",
            "fontSize": 30,
            "height": 60,
            "margin": 20,
            "padding": 10,
            "textAlign": "center",
            "width": "auto",
          }
        }
      >
        Follower
      </Text>
    </TouchableOpacity>
  </ScrollViewMock>
</View>
`;
